lib/hammer_cli_csv/content_hosts.rb

```diff
@@ -25,111 +25,53 @@ def export
-                  operatingsystem, architecture, sockets, ram, cores, sla, products, subscriptions]
-        end
-      end
-
-      def export_foretello(csv)
+      def export_katello(csv)
         @api.resource(:organizations).call(:index, {:per_page => 999999})['results'].each do |organization|
           next if option_organization && organization['name'] != option_organization
 
-          @api.resource(:systems).call(:index, {
+          @api.resource(:hosts).call(:index, {
               'per_page' => 999999,
               'organization_id' => foreman_organization(:name => organization['name'])
           })['results'].each do |host|
-            host = @api.resource(:systems).call(:show, {
-                'id' => host['uuid'],
-                'fields' => 'full'
+            host = @api.resource(:hosts).call(:show, {
+                'id' => host['id']
             })
+            host['facts'] ||= {}
 
             name = host['name']
             organization_name = organization['name']
-            environment = host['environment']['label']
-            contentview = host['content_view']['name']
-            hostcollections = CSV.generate do |column|
-              column << host['hostCollections'].collect do |hostcollection|
-                hostcollection['name']
-              end
+            if host['content_facet_attributes']
+              environment = host['content_facet_attributes']['lifecycle_environment']['name']
+              contentview = host['content_facet_attributes']['content_view']['name']
+              hostcollections = export_column(host['content_facet_attributes'], 'host_collections', 'name')
+            else
+              environment = nil
+              contentview = nil
+              hostcollections = nil
             end
-            hostcollections.delete!("\n")
-            virtual = host['facts']['virt.is_guest'] == 'true' ? 'Yes' : 'No'
-            hypervisor_host = host['virtual_host'].nil? ? nil : host['virtual_host']['name']
-            operatingsystem = "#{host['facts']['distribution.name']} " if host['facts']['distribution.name']
-            operatingsystem += host['facts']['distribution.version'] if host['facts']['distribution.version']
-            architecture = host['facts']['uname.machine']
-            sockets = host['facts']['cpu.cpu_socket(s)']
-            ram = host['facts']['memory.memtotal']
-            cores = host['facts']['cpu.core(s)_per_socket'] || 1
+            virtual = host['facts']['virt::is_guest'] == 'true' ? 'Yes' : 'No'
+            hypervisor_host = host['subscription_facet_attributes']['virtual_host'].nil? ? nil : host['subscription_facet_attributes']['virtual_host']['name']
+            operatingsystem = "#{host['facts']['distribution::name']} " if host['facts']['distribution::name']
+            operatingsystem += host['facts']['distribution::version'] if host['facts']['distribution::version']
```

Comment:
you may want the space to precede `distribution::version`, that way it's only added when needed.

Also I would declare as `' ' + host['facts']['distribution::version']` just to make it really obvious that a space is being added